app/Models/Champion.php

```diff
@@ -38,37 +54,82 @@ public function deleteItem($name)
             {
                 $this->decreaseArmor($item->armor);
                 $this->decreaseHp($item->hp);
+                $this->decreaseMagicResist($item->magicResist);
                 unset($this->items[$key]);
                 return true;
             }
         }
         return false;
     }
+
+    public function addLevel()
+    {
+        $this->level = $this->level + 1;
+        $armor = $this->armorGrowth * 1;
+        $magicResist = $this->magicResistGrowth *1;
+        $hp = $this->hpGrowth *1;
+        $this->addArmor($armor);
+        $this->addMagicResist($magicResist);
+        $this->addHp($hp);
+        $this->experience = $this->levelTreshold[$this->level][0];
+    }
+
+    public function setLevel(int $level, $experience = null)
+    {
+        $level = $level - $this->level;

```

Comment:
```suggestion
        $level -= $this->level;

```

Reply:
$level tutaj to nie konkretny poziom, tylko roznica poziomow, wiec $levelDiff chyba lepszy